pallets/api/src/fungibles/mod.rs

```diff
@@ -150,11 +192,11 @@ pub mod pallet {
 			spender: AccountIdOf<T>,
 			value: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin.clone())
+			let owner = ensure_signed(origin.clone())
 				.map_err(|e| e.with_weight(Self::weight_approve(0, 0)))?;
-			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
-			let spender = T::Lookup::unlookup(spender);
-			let id: AssetIdParameterOf<T> = id.into();
+			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &owner, &spender);
+			let spender_unlookup = T::Lookup::unlookup(spender.clone());
```

Comment:
`unlookup` returns a `Source` -> `spender_source` would be a better name.

Reply:
see commit `refactor: optimization pallet fungibles`